docs/modules/ROOT/pages/erc20.adoc

```diff
@@ -188,29 +211,34 @@ mod MyToken {
         let name = 'MyToken';
         let symbol = 'MTK';
 
-        let mut unsafe_state = ERC20::unsafe_new_contract_state();
-        ERC20::InternalImpl::initializer(ref unsafe_state, name, symbol);
+        self.erc20.initializer(name, symbol);
+        self.erc20._mint(recipient, initial_supply);
     }
 
-    /// This is a standalone function for brevity.
-    /// It's recommended to create an implementation of IERC20
-    /// to ensure that the contract exposes the entire ERC20 interface.
-    /// See the previous example.
     #[external(v0)]
-    fn decimals(self: @ContractState) -> u8 {
-        self._decimals.read()
+    impl ERC20MetadataImpl of interface::IERC20Metadata<ContractState> {
+        fn name(self: @ContractState) -> felt252 {
+            self.erc20.name()
+        }
+
+        fn symbol(self: @ContractState) -> felt252 {
+            self.erc20.symbol()
+        }
+
+        fn decimals(self: @ContractState) -> u8 {
+            self.MyToken_decimals.read()
+        }
     }
 
     #[generate_trait]
     impl InternalImpl of InternalTrait {
         fn _set_decimals(ref self: ContractState, decimals: u8) {
-            self._decimals.write(decimals);
+            self.MyToken_decimals.write(decimals);
         }
     }
 }
 ----
 
 This contract expects a `decimals` argument in the constructor and uses an internal function to write the decimals to storage.
-Note that the `_decimals` state variable must be stored in the local contract's storage because this variable does not exist in the Contracts for Cairo library.
-It's important to include the correct logic in the exposed `decimals` method and to NOT use the Contracts for Cairo `decimals` implementation in this specific case.
-The library's `decimals` implementation does not read from storage and will return `18`.
+Note that the `MyToken_decimals` state variable must be stored in the local contract's storage because this variable does not exist in the Contracts for Cairo library.
```

Comment:
```suggestion
Note that the `MyToken_decimals` state variable must be defined in the contract's storage because this variable does not exist in the component offered by OpenZeppelin Contracts for Cairo.
```